packages/vite/src/node/plugins/css.ts

```diff
@@ -223,7 +223,8 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
         const thisModule = moduleGraph.getModuleById(id)
         if (thisModule) {
           // CSS modules cannot self-accept since it exports values
-          const isSelfAccepting = !modules && !inlineRE.test(id)
+          const isSelfAccepting =
+            !modules && !inlineRE.test(id) && !htmlProxyRE.test(id)
```

Comment:
it make `index.html?html-proxy&direct&0.css` can touch `index.html` update?

Reply:
When `isSelfAccepting` is `true`, it is handled by here. (I'm not sure if this answers your question)
https://github.com/vitejs/vite/blob/bb603d3d66506deb353e4d6346d5e8361fdb9e62/packages/vite/src/client/client.ts#L76-L93

Reply:
https://github.com/vitejs/vite/blob/bb603d3d66/packages/vite/src/node/server/hmr.ts#L219-L234 

I think this is make `index.html` update 🤔

Reply:
I might got it. My commit message is misleading.
That commit fixes hmr when editing a file imported by html.

For example, editing `foo.css` was not working and it was fixed by that commit.
```html
<style> @import './foo.css' </style>
```
Editing the style tag (which I think you are saying about?) was working before that commit.

Reply:
yes, thanks for your response